tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+
+So, now that we have a game plan, let's implement it! In the `test_NftMarketplace.js` file, we can add the following code:
+
+```js
+const NftMarketplace = artifacts.require("NftMarketplace");
+const DizzyDragons = artifacts.require("DizzyDragons");
+
+contract("NftMarketplace", (accounts) => {
+  const tokenUri =
+    "https://gateway.pinata.cloud/ipfs/QmTCib5LvSrb7sshLhLvzmV7wdSdmSt3yjB4dqQaA58Td9";
+  let nftMarketplace;
+  let dizzyDragonsNft;
+  let mintedNft;
+
+  beforeEach(async () => {
+    // Deploy the marketplace
+    nftMarketplace = await NftMarketplace.deployed();
+    // Deploy a Dizzy Dragon NFT
+    dizzyDragonsNft = await DizzyDragons.deployed();
+    // Mint Daizy the Dizzy Dragon NFT
+    mintedNft = await dizzyDragonsNft.mint(tokenUri);
+  });
+
+  // TODO: Add tests here
+});
+```
+
+!!! note
+    You don't have to import Mocha or Web3 as both packages comes out-of-the-box with Truffle!
+
+Now we're ready to start writing our tests!
+
+### Test Minting NFTs {: #test-minting-nfts }
+
+For our first test, let's make sure that we are minting our new Dizzy Dragon NFT as expected. We'll use the event logs from the transaction to ensure that the `NftMinted` event of the `DizzyDragons` contract has been emitted. We'll also test that the token ID of the NFT is `1`, as we are deploying the `DizzyDragons` contract fresh before each test, so it should always be the first NFT minted.
+
+In place of the `// TODO: Add tests here` comment, you can add the following test:
+
+```js
+  it("should mint a new Dizzy Dragon NFT", async () => {
+      // Access the logs of the mint transaction
+      // Remember: mintedNft was created in the beforeEach function
+      const nftMintedLog = mintedNft.logs[2];
+
+      const event = nftMintedLog.event;
+      const tokenId = nftMintedLog.args[0].toNumber();
+
+      // Use Mocha's assert to test that the NftMinted event was emitted
+      // and the token ID of the NFT is 1
+      assert.equal(event, "NftMinted");
+      assert.equal(tokenId, 1);
+  });
+```
+
+Assuming your [Moonbeam development node is up and running](#start-development-node), you can run the test with the following command:
+
+```
+npx truffle test --network dev
+```
+
+### Test Listing NFTs {: #test-listing-nfts }
+
+For our next test, we're going to test that we can successfully list our freshly minted NFT using the `listNft` function of the `NftMarketplace` contract. So, again we'll use our event logs to test that the `NftListed` event has been emitted along with the correct state variables such as the seller and token ID.
+
+After our first test, we can go ahead and the following test:
+
+```js
+  it("should list a new Dizzy Dragon NFT", async () => {
+    // Access the logs of the mint transaction
```

Comment:
I would directly retrieve the address so you can put that in the comment.

`// Access the logs of the mint transaction to get contract address`